File: src/main/java/com/rapidprototyping/manbot/model/user/UserType.java
package com.rapidprototyping.manbot.model.user;

/**
 *
 * @author Pavan C (pavan407)
 */
public enum UserType
{
    LEARNER, INTERMEDIATE, ADVANCED, MENTOR, ADMIN;

    public boolean greaterThan(UserType other)
    {
        return ordinal() > other.ordinal();
    }

    public boolean lessThan(UserType other)
    {
        return ordinal() < other.ordinal();
    }
}

File: bot/src/main/kotlin/com/manbot/user/UserType.kt
package com.manbot.user

/**
 * @author Pavan C. (pavan407)
 */
enum class UserType
{
    LEARNER, INTERMEDIATE, ADVANCED, MENTOR, ADMIN;

    fun greaterThan(other: UserType) = compareTo(other) > 0
    fun greaterThanOrEqualTo(other: UserType) = compareTo(other) >= 0

    fun lessThan(other: UserType) = compareTo(other) < 0
    fun lessThanOrEqualTo(other: UserType) = compareTo(other) <= 0
}
